tests/snos.rs

```diff
@@ -1,45 +1,55 @@
 mod common;
+
+use blockifier::state::state_api::State;
+use common::{
+    initial_state, prepare_os_test, utils::check_output_vs_python, utils::print_a_hint,
+    TestingContext,
+};
+
 use cairo_vm::cairo_run::{cairo_run, CairoRunConfig};
-use cairo_vm::felt::Felt252;
 use cairo_vm::hint_processor::builtin_hint_processor::builtin_hint_processor_definition::{
     BuiltinHintProcessor, HintFunc,
 };
-use cairo_vm::hint_processor::builtin_hint_processor::hint_utils::get_integer_from_var_name;
-use cairo_vm::hint_processor::hint_processor_definition::HintReference;
-use cairo_vm::serde::deserialize_program::ApTracking;
-use cairo_vm::types::exec_scope::ExecutionScopes;
-use cairo_vm::vm::{errors::hint_errors::HintError, vm_core::VirtualMachine};
-use common::{check_output_vs_python, compile_contracts};
-use snos::SnOsRunner;
-use std::collections::HashMap;
+
+use snos::DEFAULT_LAYOUT;
+use snos::{state::SharedState, SnOsRunner};
+use starknet_api::block::BlockNumber;
+use starknet_api::hash::StarkFelt;
+use starknet_api::stark_felt;
+use starknet_api::transaction::Calldata;
+
 use std::fs;
 use std::rc::Rc;
 
 use rstest::*;
 
-// Create the function that implements the custom hint
-fn print_a_hint(
-    vm: &mut VirtualMachine,
-    _exec_scopes: &mut ExecutionScopes,
-    ids_data: &HashMap<String, HintReference>,
-    ap_tracking: &ApTracking,
-    _constants: &HashMap<String, Felt252>,
-) -> Result<(), HintError> {
-    let a = get_integer_from_var_name("a", vm, ids_data, ap_tracking)?;
-    println!("{}", a);
-    Ok(())
+#[rstest]
+fn snos_ok(_initial_state: TestingContext) {
+    let snos_runner = SnOsRunner::new(DEFAULT_LAYOUT.to_string(), "build/os_debug.json");
+    let _runner_res = snos_runner.run();
+}
+
+#[rstest]
+fn shared_state(initial_state: TestingContext) {
+    let mut shared_state = SharedState::new(
+        initial_state.block_context.chain_id,
+        Box::new(initial_state.state),
+    );
+    let commitment = shared_state.apply_diff(BlockNumber(1));
+
+    assert_eq!(
+        stark_felt!("473010ec333f16b84334f9924912d7a13ce8296b0809c2091563ddfb63011d"),
```

Comment:
where does that come from?

Reply:
this is the equivalent root for the python OS tests, will add comment